bees/emailserverbee/emailserverbeefactory.go

```diff
@@ -0,0 +1,199 @@
+func (factory *EmailServerBeeFactory) LogoColor() string {
+	return "#00bbff"
+}
+
+// Options returns the options available to configure this Bee.
+func (factory *EmailServerBeeFactory) Options() []bees.BeeOptionDescriptor {
+	hostname, _ := os.Hostname()
+
+	opts := []bees.BeeOptionDescriptor{
+		{
+			Name:        "address",
+			Description: "Which addr to listen on, eg: 0.0.0.0:25",
+			Type:        "address",
+			Mandatory:   true,
+		},
+		{
+			Name:        "allowedHosts",
+			Description: "AllowedHosts lists which hosts to accept email for.",
+			Type:        "[]string",
+			Default:     hostname,
+			Mandatory:   false,
+		},
+		{
+			Name:        "hostname",
+			Description: "Hostname will be used in the server's reply to HELO/EHLO. If TLS enabled make sure that the Hostname matches the cert.",
+			Default:     hostname,
+			Type:        "string",
+			Mandatory:   false,
+		},
+		{
+			Name:        "maxSize",
+			Description: "MaxSize is the maximum size of an email that will be accepted for delivery.",
+			Default:     1024 * 1024 * 10,
+			Type:        "int64",
+			Mandatory:   false,
+		},
+		{
+			Name:        "timeout",
+			Description: "Timeout specifies the connection timeout in seconds. Defaults to 30",
+			Default:     30,
+			Type:        "int",
+			Mandatory:   false,
+		},
+		{
+			Name:        "maxClients",
+			Description: "MaxClients controls how many maxiumum clients we can handle at once.",
+			Default:     10,
+			Type:        "int",
+			Mandatory:   false,
+		},
+		{
+			Name:        "startTLSOn",
+			Description: "StartTLSOn should we offer STARTTLS command. Cert must be valid.",
+			Default:     false,
+			Type:        "bool",
+			Mandatory:   false,
+		},
+		{
+			Name:        "tlsAlwaysOn",
+			Description: "TLSAlwaysOn run this server as a pure TLS server, i.e. SMTPS",
+			Default:     false,
+			Type:        "bool",
+			Mandatory:   false,
+		},
+		{
+			Name:        "privateKeyFile",
+			Description: "PrivateKeyFile path to cert private key in PEM format",
+			Default:     "",
+			Type:        "string",
+			Mandatory:   false,
+		},
+		{
+			Name:        "publicKeyFile",
+			Description: "PublicKeyFile path to cert (public key) chain in PEM format.",
+			Default:     "",
+			Type:        "string",
+			Mandatory:   false,
+		},
+	}
+	return opts
+}
+
+// Events describes the available events provided by this Bee.
+func (factory *EmailServerBeeFactory) Events() []bees.EventDescriptor {
+	events := []bees.EventDescriptor{
+		{
+			Namespace:   factory.Name(),
+			Name:        "received",
+			Description: "An Email was received",
+			Options: []bees.PlaceholderDescriptor{
+				{
+					Name:        "sender",
+					Description: "The Email sender",
+					Type:        "string",
+				},
+				{
+					Name:        "remote_ip",
+					Description: "The IP from the remote-server",
+					Type:        "string",
+				},
+				{
+					Name:        "recipients",
+					Description: "The recipients",
+					Type:        "string[]",
```

Comment:
This should read []string, as we decided to stick with Go's conventions. It currently hasn't much of an effect within Beehive, but the admin interface will make use of that information, soon.